src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnector.java

```diff
@@ -0,0 +1,167 @@
+        PropertyType.STRING).build());
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  public TableList listTables() {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint(), OBJECT_TABLE_LIST);
+    String accessToken = null;
+    ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(config);
+    try {
+      accessToken = serviceNowTableAPIClient.getAccessToken();
+    } catch (OAuthSystemException | OAuthProblemException e) {
+      e.printStackTrace();
+    }
+    requestBuilder.setAuthHeader(accessToken);
+    requestBuilder.setAcceptHeader(MediaType.APPLICATION_JSON);
+    requestBuilder.setContentTypeHeader(MediaType.APPLICATION_JSON);
+    RestAPIResponse apiResponse = null;
+    apiResponse = serviceNowTableAPIClient.executeGet(requestBuilder.build());
+    if (!apiResponse.isSuccess()) {
+      LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+      throw new RuntimeException();
+    } else {
+      String response = null;
+      response = apiResponse.getResponseBody();
+      Gson gson = new Gson();
+      TableList tables = gson.fromJson(response, TableList.class);
+      return tables;
+    }
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(io.cdap.plugin.common.ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String tableName = connectorSpecRequest.getPath();
+    if (tableName != null) {
+      properties.put(ServiceNowConstants.PROPERTY_TABLE_NAME, tableName);
+    }
+    SourceQueryMode mode = SourceQueryMode.TABLE;
+    SourceValueType valueType = SourceValueType.SHOW_DISPLAY_VALUE;
+    List<ServiceNowTableInfo> tableInfo = ServiceNowInputFormat.fetchTableInfo(mode, config, tableName,
+                                                                               null, valueType,
+                                                                               null, null);
+    Schema schema = tableInfo.stream().findFirst().get().getSchema();
+    specBuilder.setSchema(schema);
+    return specBuilder.addRelatedPlugin(new PluginSpec(ServiceNowConstants.PLUGIN_NAME, BatchSource.PLUGIN_TYPE,
+                                                       properties)).
+      addRelatedPlugin(new PluginSpec(ServiceNowConstants.PLUGIN_NAME, BatchSink.PLUGIN_TYPE, properties)).build();
+  }
+
+
+  private String getErrorMessage(String responseBody) {
+    try {
+      Gson gson = new Gson();
```

Comment:
Use the class level variable for `gson` after defining it.

Reply:
Done